Carthage support

Hey, do you plan on supporting Carghage? It would really help :) Currently all I get is a bunch of errors about failed tests.

Here is my Cartfile:

```
github "p2/OAuth2" ~> 2.2
```

Building for iOS works for me:
```
$ carthage build --platform iOS
*** Building scheme "OAuth2iOS" in OAuth2.xcworkspace
```

However, I wasn't able to build for OS X out of the box:

```
$ carthage build --platform Mac
*** Building scheme "OAuth2OSX" in OAuth2.xcworkspace
** BUILD FAILED **

.../Carthage/Checkouts/OAuth2/Tests/OAuth2AuthRequest_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ClientCredentials_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2CodeGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ImplicitGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2DynReg_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2PasswordGrant_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2RefreshToken_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2AuthRequest_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ClientCredentials_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2CodeGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ImplicitGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2DynReg_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2PasswordGrant_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2RefreshToken_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2AuthRequest_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ClientCredentials_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2CodeGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ImplicitGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2DynReg_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2PasswordGrant_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2RefreshToken_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2AuthRequest_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ClientCredentials_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2CodeGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2ImplicitGrant_tests.swift:24:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2DynReg_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2PasswordGrant_tests.swift:23:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2RefreshToken_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing
.../Carthage/Checkouts/OAuth2/Tests/OAuth2_tests.swift:27:8: error: module 'OAuth2' was not compiled for testing

A shell task failed with exit code 65:
** BUILD FAILED **

The following build commands failed:
	CompileSwift normal x86_64 .../Carthage/Checkouts/OAuth2/Tests/OAuth2AuthRequest_tests.swift
	CompileSwift normal x86_64 .../Carthage/Checkouts/OAuth2/Tests/OAuth2ClientCredentials_tests.swift
	CompileSwift normal x86_64 .../Carthage/Checkouts/OAuth2/Tests/OAuth2CodeGrant_tests.swift
	CompileSwift normal x86_64 .../Carthage/Checkouts/OAuth2/Tests/OAuth2ImplicitGrant_tests.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(5 failures)
```

I am able to build for Mac if I select "Enable Testability" for the OAuth2OSX Release target:

<img width="597" alt="screen shot 2016-03-27 at 1 24 32 am" src="https://cloud.githubusercontent.com/assets/266156/14064511/bb7079c4-f3ba-11e5-885b-fffab019cc58.png">

--- a/OAuth2.xcodeproj/xcshareddata/xcschemes/OAuth2OSX.xcscheme
+++ b/OAuth2.xcodeproj/xcshareddata/xcschemes/OAuth2OSX.xcscheme
@@ -22,7 +22,7 @@
          </BuildActionEntry>
          <BuildActionEntry
             buildForTesting = "YES"
-            buildForRunning = "YES"
+            buildForRunning = "NO"
             buildForProfiling = "NO"
             buildForArchiving = "NO"
             buildForAnalyzing = "YES">

